### PR TITLE
fix(cache): fix migration and catch error class in CI

### PR DESCRIPTION
**Pull request by cschmatzler**
Fun.
#9499 added a new migration with a non-constant `default`. It worked locally (on an empty dev database), it worked in `mix test`, it worked in CI.
It didn't work when deploying.

SQLite allows non-constant defaults in `ALTER TABLE` on _empty_ tables, not on non-empty ones. All of our tests run on an empty database.

This PR fixes the migration and also adds a new CI step that tests migrations on a non-empty database.

### CODE REVIEW

File: .github/workflows/cache.yml [hunk 1]
```diff
@@ -122,6 +122,45 @@ jobs:
...
         run: MIX_ENV=prod mix compile --warnings-as-errors
 
+  migration-safety:
+    name: Migration Safety
+    runs-on: namespace-profile-default
+    timeout-minutes: 15
+    if: github.event.pull_request.draft == false || github.event_name != 'pull_request'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Restore Mix Cache
```
Comment (fortmarek) [inline .github/workflows/cache.yml]: I would suggest aligning with the approach here, so this job doesn't become unnecessarily slow: https://github.com/tuist/tuist/pull/9530

File: .github/workflows/cache.yml [hunk 2]
```diff
@@ -122,6 +122,45 @@ jobs:
...
+          path: |
+            cache/deps
+            cache/_build
+          key: cache-mix-${{ hashFiles('cache/mix.lock') }}
+      - uses: jdx/mise-action@v3.2.0
+        with:
+          cache_key: "{{default}}-{{env.NSC_BASE_IMAGE_REF_ID}}"
+          working_directory: cache
+      - name: Install dependencies
+        run: mise run install
```
Comment (fortmarek) [inline .github/workflows/cache.yml]: let's use `mix deps.get` instead

File: .github/workflows/cache.yml [hunk 3]
```diff
@@ -122,6 +122,45 @@ jobs:
...
+      - name: Fetch main branch
+        run: git fetch origin main --depth=1
+      - name: Test migrations against seeded database
+        run: |
+          NEW_MIGRATIONS=$(git diff --name-only --relative origin/main -- priv/repo/migrations/ | grep '\.exs$' | sort || true)
+
+          if [ -z "$NEW_MIGRATIONS" ]; then
+            echo "No new migrations found, skipping."
+            exit 0
+          fi
```
Comment (fortmarek) [inline .github/workflows/cache.yml]: instead, can't we run this job selectively based on path changes at the GHA level?
